pkg/util/liveconfig/manager.go

```diff
@@ -21,10 +21,20 @@ type Manager interface {
 	DefaultInstallerPullSpecOverride(context.Context) string
 }
 
-type dev struct{}
+type dev struct {
+	location              string
+	managedClustersClient containerservice.ManagedClustersClient
+
+	hiveCredentialsMutex *sync.RWMutex
```

Comment:
the pointer isn't need here, (will need to probably change prod too), because we are only using the pointers to that struct and using it as an interface. that allows to not instantiate the mutex

Reply:
Changed.

Reply:
has it ?

Reply:
Well, funny story... it was, but when I switched back to not moving hive -> prod, etc. (because of mock-hell) I reverted it by mistake. Fixed in the latest push.